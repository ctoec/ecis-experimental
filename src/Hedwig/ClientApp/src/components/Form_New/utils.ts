import moment from 'moment';

export function parseDateChange(event: React.ChangeEvent<any>) {
	return event.target.value ? moment.utc(parseInt(event.target.value, 10)).toDate() : null;
}
