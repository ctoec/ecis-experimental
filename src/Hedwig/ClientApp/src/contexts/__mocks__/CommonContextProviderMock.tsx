import React from 'react';
import { Router } from 'react-router-dom';
import UserContext from '../User/UserContext';
import ReportingPeriodContext from '../ReportingPeriod/ReportingPeriodContext';
import { User, Region, ReportingPeriod, FundingSource } from '../../generated';
import emptyGuid from '../../utils/emptyGuid';
import { createMemoryHistory, History } from 'history';

export const defaultUser: User = {
	id: 1,
	wingedKeysId: emptyGuid(),
	firstName: 'Minerva',
	lastName: 'McGonagall',
	orgPermissions: [
		{
			organizationId: 1,
			organization: {
				id: 1,
				name: "Children's Adventure Center",
				sites: [
					{
						id: 1,
						name: "Children's Adventure Center",
						organizationId: 1,
						region: Region.East,
						titleI: false,
					},
				],
			},
			id: 1,
			userId: 1,
		},
	],
	sitePermissions: [],
};

export const defaultCdcReportingPeriods: ReportingPeriod[] = [
	{
		id: 1,
		type: FundingSource.CDC,
		period: new Date('2019-03-01'),
		periodStart: new Date('2019-03-01'),
		periodEnd: new Date('2019-03-31'),
		dueAt: new Date('2019-04-15'),
	},
	{
		id: 2,
		type: FundingSource.CDC,
		period: new Date('2019-04-01'),
		periodStart: new Date('2019-04-01'),
		periodEnd: new Date('2019-04-30'),
		dueAt: new Date('2019-05-15'),
	},
	{
		id: 3,
		type: FundingSource.CDC,
		period: new Date('2019-05-01'),
		periodStart: new Date('2019-05-01'),
		periodEnd: new Date('2019-05-31'),
		dueAt: new Date('2019-06-15'),
	},
];

type CommonContextProviderMockProps = {
	user?: User;
	cdcReportingPeriods?: ReportingPeriod[];
	history?: History<any>;
};

const CommonContextProviderMock: React.FC<CommonContextProviderMockProps> = ({
	children,
	user = defaultUser,
	cdcReportingPeriods = defaultCdcReportingPeriods,
	history = createMemoryHistory(),
}) => {
	return (
		<UserContext.Provider value={{ user }}>
			<ReportingPeriodContext.Provider value={{ cdcReportingPeriods }}>
				<Router history={history}>{children}</Router>
			</ReportingPeriodContext.Provider>
		</UserContext.Provider>
	);
};

export default CommonContextProviderMock;
