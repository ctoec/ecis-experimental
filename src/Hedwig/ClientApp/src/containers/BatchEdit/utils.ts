import { Enrollment } from "../../generated";
import { hasValidationErrors } from "../../utils/validations";

export const hasChildInfoSectionErrors = (enrollment: Enrollment) => 	hasValidationErrors(enrollment.child, [
		'birthdate',
		'birthCertificateId',
		'birthtown',
		'birthstate',
		'hispanicOrLatinxEthnicity',
		'americanIndianOrAlaskaNative',
		'asian',
		'blackOrAfricanAmerican',
		'nativeHawaiianOrPacificIslander',
		'white',
		'gender',
	]);

	export const hasFamilyInfoSectionErrors = (enrollment: Enrollment) => hasValidationErrors(enrollment.child?.family, undefined, true);

	export const hasFamilyIncomeSectionErrors = (enrollment: Enrollment) => hasValidationErrors(enrollment.child?.family, ['determinations']);

	export const hasEnrollmentFundingSectionErrors = (enrollment: Enrollment) => 
		hasValidationErrors(enrollment, ['entry']) ||
		hasValidationErrors(enrollment.child, ['c4KCertificateFamilyId', 'c4KCertificates']);


