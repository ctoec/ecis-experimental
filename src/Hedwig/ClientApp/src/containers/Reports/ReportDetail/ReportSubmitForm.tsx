import React, { useState, useEffect, useContext, useReducer } from 'react';
import {
	CdcReport,
	ApiOrganizationsOrgIdReportsIdPutRequest,
	ApiOrganizationsOrgIdEnrollmentsGetRequest,
} from '../../../generated';
import useApi, { Mutate, ApiError } from '../../../hooks/useApi';
import UserContext from '../../../contexts/User/UserContext';
import { Button, TextInput, ChoiceList, FieldSet, ErrorBoundary } from '../../../components';
import AppContext from '../../../contexts/App/AppContext';
import currencyFormatter from '../../../utils/currencyFormatter';
import parseCurrencyFromString from '../../../utils/parseCurrencyFromString';
import { getIdForUser, activeC4kFundingAsOf } from '../../../utils/models';
import UtilizationTable from './UtilizationTable';
import AlertContext from '../../../contexts/Alert/AlertContext';
import { useHistory } from 'react-router';
import { DeepNonUndefineable } from '../../../utils/types';
import {
	useFocusFirstError,
	serverErrorForField,
	isBlockingValidationError,
} from '../../../utils/validations';
import usePromiseExecution from '../../../hooks/usePromiseExecution';
import { reportSubmittedAlert, reportSubmitFailAlert } from '../../../utils/stringFormatters';
import pluralize from 'pluralize';
import { validationErrorAlert } from '../../../utils/stringFormatters/alertTextMakers';
import { FormReducer, formReducer, updateData } from '../../../utils/forms/form';

export type ReportSubmitFormProps = {
	report: DeepNonUndefineable<CdcReport>;
	mutate: Mutate<CdcReport>;
	error: ApiError | null;
	canSubmit: boolean;
};

export default function ReportSubmitForm({
	report,
	mutate,
	error,
	canSubmit,
}: ReportSubmitFormProps) {
	const history = useHistory();

	const { user } = useContext(UserContext);

	const [_report, updateReport] = useReducer<FormReducer<DeepNonUndefineable<CdcReport>>>(
		formReducer,
		report
	);
	const updateFormData = updateData<DeepNonUndefineable<CdcReport>>(updateReport);
	const {
		submittedAt,
		accredited,
		c4KRevenue,
		retroactiveC4KRevenue,
		familyFeesRevenue,
		comment,
	} = _report;

	const { invalidateCache: invalidateAppCache } = useContext(AppContext);
	const { getAlerts, setAlerts } = useContext(AlertContext);
	const alerts = getAlerts();

	const params: ApiOrganizationsOrgIdReportsIdPutRequest = {
		id: report.id || 0,
		orgId: getIdForUser(user, 'org'),
	};

	const enrollmentParams: ApiOrganizationsOrgIdEnrollmentsGetRequest = {
		orgId: getIdForUser(user, 'org'),
		include: ['fundings'],
		startDate: report.reportingPeriod.periodStart,
		endDate: report.reportingPeriod.periodEnd,
		asOf: submittedAt || undefined,
	};

	const [, , allEnrollments] = useApi(
		api => api.apiOrganizationsOrgIdEnrollmentsGet(enrollmentParams),
		[user, report]
	);

	const [care4KidsCount, setCare4KidsCount] = useState(0);

	useEffect(() => {
		if (allEnrollments) {
			var c4kFundedEnrollments = allEnrollments.filter(
				enrollment => !!activeC4kFundingAsOf(enrollment.fundings, submittedAt || undefined)
			);
			var childIds: string[] = [];
			c4kFundedEnrollments.forEach(enrollment => {
				const childId = enrollment.childId;
				if (childIds.indexOf(childId) < 0) {
					childIds.push(childId);
				}
			});
			setCare4KidsCount(childIds.length);
		}
	}, [allEnrollments]);

	// set up form state
	useFocusFirstError([error]);
	const [hasAlertedOnError, setHasAlertedOnError] = useState(false);
	useEffect(() => {
		if (error && !hasAlertedOnError) {
			if (!isBlockingValidationError(error)) {
				throw new Error(error.title || 'Unknown api error');
			}
			setAlerts([validationErrorAlert]);
		}
	}, [error, hasAlertedOnError]);

	function _onSubmit() {
		return mutate(api =>
			api.apiOrganizationsOrgIdReportsIdPut({
				...params,
				cdcReport: { ..._report },
			})
		).then(res => {
			if (res) {
				const newAlert = reportSubmittedAlert(report.reportingPeriod);
				const newAlerts = [...alerts, newAlert];
				setAlerts(newAlerts);
				invalidateAppCache(); // Updates the count of unsubmitted reports in the nav bar
				history.push('/reports', newAlerts);
			}
		});
	}
	const { isExecuting: isMutating, setExecuting: onSubmit } = usePromiseExecution(_onSubmit);

	return (
		<ErrorBoundary alertProps={reportSubmitFailAlert}>
			{report.submittedAt && (
				<p>
					<b>Submitted:</b> {report.submittedAt.toLocaleDateString()}{' '}
				</p>
			)}
			<ChoiceList
				type="check"
				id="accredited"
				name="accredited"
				legend="Accredited"
				disabled={!!report.submittedAt}
				selected={accredited ? ['accredited'] : undefined}
				options={[
					{
						text: 'Accredited',
						value: 'accredited',
					},
				]}
				onChange={updateFormData((_, e) => e.target.checked)}
				className="margin-bottom-5"
			/>
			<UtilizationTable {...{ ..._report, accredited }} />
			<form className="usa-form" onSubmit={onSubmit} noValidate autoComplete="off">
				<h2>Other Revenue</h2>
				<FieldSet id="other-revenue" legend="Other Revenue">
					<TextInput
						type="input"
						id="c4k-revenue"
						name="c4KRevenue"
						label={
							<React.Fragment>
								<span className="text-bold">Care 4 Kids</span>
								<span>
									{' '}
									({care4KidsCount} {pluralize('kid', care4KidsCount)} receiving subsidies)
								</span>
							</React.Fragment>
						}
						defaultValue={currencyFormatter(c4KRevenue)}
						onChange={updateFormData(parseCurrencyFromString)}
						onBlur={event =>
							(event.target.value = c4KRevenue !== null ? currencyFormatter(c4KRevenue) : '')
						}
						optional={true}
						disabled={!!report.submittedAt}
						status={serverErrorForField(
							hasAlertedOnError,
							setHasAlertedOnError,
							'report.c4krevenue',
							error,
							'This information is required for the report'
						)}
					/>
					<ChoiceList
						type="check"
						id="c4k-includes-retroactive"
						legend="Includes retroactive payment"
						name="retroactiveC4KRevenue"
						selected={retroactiveC4KRevenue ? ['retroactiveC4KRevenue'] : undefined}
						onChange={updateFormData((_, e) => e.target.checked)}
						disabled={!!report.submittedAt}
						options={[
							{
								text: 'Includes retroactive payment for past months',
								value: 'retroactiveC4KRevenue',
							},
						]}
					/>
					<TextInput
						type="input"
						id="family-fees-revenue"
						name="familyFeesRevenue"
						label={<span className="text-bold">Family Fees</span>}
						defaultValue={currencyFormatter(familyFeesRevenue)}
						onChange={updateFormData(parseCurrencyFromString)}
						onBlur={event =>
							(event.target.value =
								familyFeesRevenue !== null ? currencyFormatter(familyFeesRevenue) : '')
						}
						disabled={!!report.submittedAt}
						status={serverErrorForField(
							hasAlertedOnError,
							setHasAlertedOnError,
							'familyfeesrevenue',
							error,
							'This information is required'
						)}
					/>
					<TextInput
						type="textarea"
						id="cdc-report-comment"
						name="comment"
						label={
							<span className="text-bold">
								Anything to share with the Office of Early Childhood about your report?
							</span>
						}
						defaultValue={comment || ''}
						onChange={updateFormData()}
						disabled={!!report.submittedAt}
						optional={true}
					/>
				</FieldSet>
				{!report.submittedAt && (
					<Button
						onClick="submit"
						text={isMutating ? 'Submitting...' : 'Submit'}
						disabled={!canSubmit || isMutating}
					/>
				)}
			</form>
		</ErrorBoundary>
	);
}
