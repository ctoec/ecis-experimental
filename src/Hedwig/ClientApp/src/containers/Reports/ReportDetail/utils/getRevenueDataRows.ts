import { CdcReport, FundingTime, Enrollment, FundingSpace, FundingSource, Age } from "../../../../generated";
import idx from "idx";
import { getReportingPeriodWeeks, getFundingSpaces, fundingSpaceSorter } from "../../../../utils/models";
import { getSplitUtilizations } from "../../../../utils/models/fundingTimeSplitUtilization";
import { DeepNonUndefineableArray } from "../../../../utils/types";
import { countFundedEnrollments, calculateRate, productOfUnknowns } from "../../../../utils/utilizationTable";

export interface RevenueDataRow  {
	key: string;
	ageGroup?: Age;
	fundingTime?: FundingTime;
	count: number;
	capacity: number;
	ftRate?: number;
	ptRate?: number;
	total: number;
	balance: number;
	weeksSplit?: {
		partWeeks: number | undefined;
		fullWeeks: number | undefined;
	};
}
export const getRevenueDataRows = (report: CdcReport) => {
	// TODO: if the space lives on the organization but the rates live on the site,
	// we need to reconsider how we handle multi site org rate calculation
	const site = idx(report, (_) => _.organization.sites[0]);
	const fundingSpaces = idx(report, (_) => _.organization.fundingSpaces);

	if (!site || !fundingSpaces) return [];

	const weeksInPeriod = getReportingPeriodWeeks(report.reportingPeriod);
	const splitTimeFundingSpaces = getFundingSpaces(
		report.organization && report.organization.fundingSpaces,
		{
			time: FundingTime.Split,
		}
	);
	const timeSplitUtilizations = getSplitUtilizations(report, splitTimeFundingSpaces, weeksInPeriod);

	const enrollments = (idx(report, (_) => _.enrollments) || []) as Enrollment[];

	const cdcFundingSpaces = (fundingSpaces as DeepNonUndefineableArray<FundingSpace>)
		.sort(fundingSpaceSorter)
		.filter((fundingSpace) => fundingSpace.source === FundingSource.CDC);

	let rows: RevenueDataRow[] = cdcFundingSpaces.map((space) => {
		const capacity = space.capacity;
		const ageGroup = space.ageGroup;
		const fundingTime = space.time;
		const count = countFundedEnrollments(enrollments, space.id);
		const ptRate = calculateRate(
			report.accredited,
			site.titleI,
			site.region,
			ageGroup,
			FundingTime.Part
		);
		const ftRate = calculateRate(
			report.accredited,
			site.titleI,
			site.region,
			ageGroup,
			FundingTime.Full
		);

		let fullWeeks: number | undefined, partWeeks: number | undefined;
		if (fundingTime === FundingTime.Split) {
			const thisUtilization = (timeSplitUtilizations || []).find((u) => u.reportId === report.id);
			fullWeeks = 0;
			partWeeks = 0;
			if (thisUtilization) {
				fullWeeks = thisUtilization.fullTimeWeeksUsed || 0;
				partWeeks = thisUtilization.partTimeWeeksUsed || 0;
			}
		} else if (fundingTime === FundingTime.Full) {
			fullWeeks = weeksInPeriod;
		} else if (fundingTime === FundingTime.Part) {
			partWeeks = weeksInPeriod;
		}

		const cappedCount = Math.min(count, capacity);
		const total =
			productOfUnknowns([cappedCount, fullWeeks, ftRate]) +
			productOfUnknowns([cappedCount, partWeeks, ptRate]);
		const paid =
			productOfUnknowns([capacity, fullWeeks, ftRate]) +
			productOfUnknowns([capacity, partWeeks, ptRate]);

		const balance = total - paid;

		return {
			key: `${ageGroup}-${fundingTime}`,
			ageGroup,
			fundingTime,
			count,
			capacity,
			ftRate,
			ptRate,
			total,
			balance,
			weeksSplit: {
				partWeeks: partWeeks,
				fullWeeks: fullWeeks,
			},
		};
	});

	const totalRow = rows.reduce(
		(total, row) => {
			return {
				...total,
				count: total.count + row.count,
				capacity: total.capacity + row.capacity,
				total: total.total + row.total,
				balance: total.balance + row.balance,
			};
		},
		{
			key: 'total',
			count: 0,
			capacity: 0,
			total: 0,
			balance: 0,
		}
	);

	rows.push(totalRow);

	return rows;
};
