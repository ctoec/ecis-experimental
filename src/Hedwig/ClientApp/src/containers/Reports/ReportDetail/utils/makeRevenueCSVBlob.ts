import { parse as parseCsv } from 'json2csv';
import { CdcReport, FundingTime } from "../../../../generated";
import { prettyAge, prettyFundingTime, getReportingPeriodWeeks } from "../../../../utils/models";
import { getRevenueDataRows } from "./getRevenueDataRows";

export const makeRevenueCSVBlob = (report: CdcReport) => {
	let revenueData = getRevenueDataRows(report);
	const totalRow = revenueData.find((row) => row.key === 'total');

	const revenueCSVRows: any[] = revenueData.map((row) => ({
		...row,
		title:
			row.key === 'total'
				? 'Total'
				: `${prettyAge(row.ageGroup)} - ${prettyFundingTime(row.fundingTime)}`,
	}));

	revenueCSVRows.push(
		{}, // empty row for final CSV formatting
		{ title: 'Other revenue' },
		{}, // empty row for final CSV formatting
		{ title: 'Care 4 Kids', total: report.c4KRevenue },
		{ title: 'Family Fees', total: report.familyFeesRevenue },
		{}, // empty row for final CSV formatting
		{
			title: 'Total revenue',
			total: (totalRow?.total || 0) + (report.c4KRevenue || 0) + (report.familyFeesRevenue || 0),
		}
	);

	const fields = [
		{
			label: 'CDC Revenue',
			value: 'title',
		},
		{
			label: 'Utilization',
			value: (row: any) =>
				row.count != undefined && row.capacity != undefined ? `${row.count} / ${row.capacity}` : '',
		},
		{
			label: 'Reimbursement rate',
			value: (row: any) => {
				let rateStr = '';
				if (row.ptRate != undefined && row.fundingTime && row.fundingTime !== FundingTime.Full)
					rateStr += `${row.ptRate} (part-time) `;
				if (row.ftRate != undefined && row.fundingTime && row.fundingTime !== FundingTime.Part)
					rateStr += `${row.ftRate} (full-time) `;
				return rateStr;
			},
		},
		{
			label: `Total (${getReportingPeriodWeeks(report.reportingPeriod)} weeks)`,
			value: 'total',
		},
		{
			label: 'Balance',
			value: (row: any) => (row.balance != undefined ? (row.balance as number).toFixed(2) : ''),
		},
	];

	const revenueCsv = parseCsv(revenueCSVRows, { fields });
	return new Blob([revenueCsv], { type: 'text/csv' });
};
