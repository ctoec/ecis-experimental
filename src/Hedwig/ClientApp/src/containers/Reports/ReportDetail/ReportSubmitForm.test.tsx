import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import 'react-dates/initialize';
import ReportSubmitForm from './ReportSubmitForm';
import CommonContextProviderMock from '../../../contexts/__mocks__/CommonContextProviderMock';
import { defaultReport } from '../../../hooks/__mocks__/useApi';
import { DeepNonUndefineable } from '../../../utils/types';
import { CdcReport } from '../../../generated';

jest.mock('../../../hooks/useApi');

afterAll(() => {
	jest.resetModules();
});

describe('ReportSubmitForm', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(
			<CommonContextProviderMock>
				<ReportSubmitForm
					report={defaultReport as DeepNonUndefineable<CdcReport>}
					mutate={() => Promise.resolve()}
					canSubmit={true}
				/>
			</CommonContextProviderMock>
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it('updates rates if accreditation is changed', () => {
		const { getByText, getByLabelText } = render(
			<CommonContextProviderMock>
				<ReportSubmitForm
					report={defaultReport as DeepNonUndefineable<CdcReport>}
					mutate={() => Promise.resolve()}
					canSubmit={true}
				/>
			</CommonContextProviderMock>
		);

		expect(getByText('Preschool').closest('tr')).toHaveTextContent('$165.32');

		fireEvent.click(getByLabelText('Accredited'));

		expect(getByText('Preschool').closest('tr')).toHaveTextContent('$126.59');
	});

	it('pretty formats currency values', () => {
		const { getByLabelText } = render(
			<CommonContextProviderMock>
				<ReportSubmitForm
					report={defaultReport as DeepNonUndefineable<CdcReport>}
					mutate={() => Promise.resolve()}
					canSubmit={true}
				/>
			</CommonContextProviderMock>
		);

		fireEvent.change(getByLabelText('Family Fees'), { target: { value: '12,34a.5' } });
		fireEvent.blur(getByLabelText('Family Fees'));

		expect(getByLabelText('Family Fees')).toHaveValue('$1,234.50');
	});
});
