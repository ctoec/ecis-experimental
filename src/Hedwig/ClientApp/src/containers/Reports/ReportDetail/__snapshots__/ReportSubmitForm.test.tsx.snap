// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportSubmitForm matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="usa-form-group "
  >
    <div
      class="margin-top-3"
    />
    <div
      class="usa-checkbox undefined"
    >
      <input
        checked=""
        class="usa-checkbox__input"
        id="accredited-accredited"
        name="accredited"
        type="checkbox"
        value="accredited"
      />
      <label
        class="usa-checkbox__label"
        for="accredited-accredited"
      >
        Accredited
      </label>
    </div>
  </div>
  <fieldset
    aria-required="true"
    class="grid-gap grid-row usa-fieldset usa-form usa-form--full-width margin-bottom-5"
    id="time-split-utilizations"
  >
    <legend
      id="fieldset-legend-time-split-utilizations"
    >
      <span
        class="usa-sr-only"
      >
        Funding time split utilizations
      </span>
    </legend>
    <div
      class="grid-gap grid-row"
    >
      <div
        class=" usa-form-group"
      >
        <label
          class="usa-label"
          for="7-lesser-weeks-used"
        >
          <span
            class="text-bold"
          >
            Infant/Toddler services were provided part time
          </span>
           
        </label>
        <input
          aria-required="true"
          class="usa-input display-inline-flex usa-input--small"
          id="7-lesser-weeks-used"
          name="splitFundingTimeUtilization"
          type="text"
          value="0"
        />
        <span
          class="display-inline-flex padding-left-1"
        >
          of 4 weeks in September.
          <span
            class="text-italic text-gray-70"
          >
             (10 part time weeks remaining)
          </span>
        </span>
      </div>
    </div>
  </fieldset>
  <h2
    class="margin-bottom-neg-205"
  >
    Monthly utilization
  </h2>
  <table
    class="oec-table oec-table--full-width"
    id="utilization-table"
  >
    <thead>
      <tr>
        <th
          aria-sort="none"
          class="oec-table__column-header"
          role="columnheader"
          scope="col"
        >
          <span
            class="oec-table__column-title"
          />
        </th>
        <th
          aria-sort="none"
          class="oec-table__column-header"
          role="columnheader"
          scope="col"
        >
          <span
            class="oec-table__column-title"
          >
            Utilization
          </span>
        </th>
        <th
          aria-sort="none"
          class="oec-table__column-header text-right"
          role="columnheader"
          scope="col"
        >
          <span
            class="oec-table__column-title"
          >
            Reimbursement rate
          </span>
        </th>
        <th
          aria-sort="none"
          class="oec-table__column-header text-right"
          role="columnheader"
          scope="col"
        >
          <span
            class="oec-table__column-title"
          >
            Total (4 weeks)
          </span>
        </th>
        <th
          aria-sort="none"
          class="oec-table__column-header text-right"
          role="columnheader"
          scope="col"
        >
          <span
            class="oec-table__column-title"
          >
            Balance
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th>
          <strong>
            Infant/Toddler
          </strong>
           – full time
        </th>
        <td
          class=" oec-table__cell--red"
        >
          1/10 spaces
        </td>
        <td
          class="text-tabular text-right"
        >
          <div>
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            />
            169.95
            <span>
               × 4 weeks
            </span>
            <div
              class="margin-left-05 width-2 display-inline-block"
            >
              (ft)
            </div>
          </div>
        </td>
        <td
          class="text-tabular text-right"
        >
          <span>
            $ 
          </span>
          <span
            style="visibility: hidden;"
          >
            0,
          </span>
          679.80
        </td>
        <td
          class="oec-table__cell--red text-tabular text-right"
        >
          <div
            class="position-relative one-half-char-right"
          >
            (
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            >
              0
            </span>
            6,118.20)
          </div>
        </td>
      </tr>
      <tr>
        <th>
          <strong>
            Infant/Toddler
          </strong>
           – part time
        </th>
        <td
          class=" oec-table__cell--red"
        >
          1/10 spaces
        </td>
        <td
          class="text-tabular text-right"
        >
          <div>
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            >
              0
            </span>
            59.50
            <span>
               × 4 weeks
            </span>
            <div
              class="margin-left-05 width-2 display-inline-block"
            >
              (pt)
            </div>
          </div>
        </td>
        <td
          class="text-tabular text-right"
        >
          <span>
            $ 
          </span>
          <span
            style="visibility: hidden;"
          >
            0,
          </span>
          238.00
        </td>
        <td
          class="oec-table__cell--red text-tabular text-right"
        >
          <div
            class="position-relative one-half-char-right"
          >
            (
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            >
              0
            </span>
            2,142.00)
          </div>
        </td>
      </tr>
      <tr>
        <th>
          <strong>
            Infant/Toddler
          </strong>
           – pt/ft split
        </th>
        <td
          class=" oec-table__cell--red"
        >
          0/10 spaces
        </td>
        <td
          class="text-tabular text-right"
        >
          <div>
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            >
              0
            </span>
            59.50
            <span>
               × 0 weeks
            </span>
            <div
              class="margin-left-05 width-2 display-inline-block"
            >
              (pt)
            </div>
          </div>
          <div>
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            />
            169.95
            <span>
               × 4 weeks
            </span>
            <div
              class="margin-left-05 width-2 display-inline-block"
            >
              (ft)
            </div>
          </div>
        </td>
        <td
          class="text-tabular text-right"
        >
          <span>
            $ 
          </span>
          <span
            style="visibility: hidden;"
          >
            000,
          </span>
          0.00
        </td>
        <td
          class="oec-table__cell--red text-tabular text-right"
        >
          <div
            class="position-relative one-half-char-right"
          >
            (
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            >
              0
            </span>
            6,798.00)
          </div>
        </td>
      </tr>
      <tr>
        <th>
          <strong>
            Preschool
          </strong>
           – full time
        </th>
        <td
          class=" oec-table__cell--red"
        >
          5/10 spaces
        </td>
        <td
          class="text-tabular text-right"
        >
          <div>
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            />
            165.32
            <span>
               × 4 weeks
            </span>
            <div
              class="margin-left-05 width-2 display-inline-block"
            >
              (ft)
            </div>
          </div>
        </td>
        <td
          class="text-tabular text-right"
        >
          <span>
            $ 
          </span>
          <span
            style="visibility: hidden;"
          />
          3,306.40
        </td>
        <td
          class="oec-table__cell--red text-tabular text-right"
        >
          <div
            class="position-relative one-half-char-right"
          >
            (
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            >
              0
            </span>
            3,306.40)
          </div>
        </td>
      </tr>
      <tr>
        <th>
          <strong>
            Preschool
          </strong>
           – part time
        </th>
        <td
          class=" oec-table__cell--red"
        >
          1/10 spaces
        </td>
        <td
          class="text-tabular text-right"
        >
          <div>
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            >
              0
            </span>
            57.85
            <span>
               × 4 weeks
            </span>
            <div
              class="margin-left-05 width-2 display-inline-block"
            >
              (pt)
            </div>
          </div>
        </td>
        <td
          class="text-tabular text-right"
        >
          <span>
            $ 
          </span>
          <span
            style="visibility: hidden;"
          >
            0,
          </span>
          231.40
        </td>
        <td
          class="oec-table__cell--red text-tabular text-right"
        >
          <div
            class="position-relative one-half-char-right"
          >
            (
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            >
              0
            </span>
            2,082.60)
          </div>
        </td>
      </tr>
      <tr>
        <th>
          <strong>
            School-age
          </strong>
           – full time
        </th>
        <td
          class=" oec-table__cell--red"
        >
          0/10 spaces
        </td>
        <td
          class="text-tabular text-right"
        >
          <div>
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            />
            113.20
            <span>
               × 4 weeks
            </span>
            <div
              class="margin-left-05 width-2 display-inline-block"
            >
              (ft)
            </div>
          </div>
        </td>
        <td
          class="text-tabular text-right"
        >
          <span>
            $ 
          </span>
          <span
            style="visibility: hidden;"
          >
            000,
          </span>
          0.00
        </td>
        <td
          class="oec-table__cell--red text-tabular text-right"
        >
          <div
            class="position-relative one-half-char-right"
          >
            (
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            >
              0
            </span>
            4,528.00)
          </div>
        </td>
      </tr>
      <tr>
        <th>
          <strong>
            School-age
          </strong>
           – part time
        </th>
        <td
          class=" oec-table__cell--red"
        >
          0/10 spaces
        </td>
        <td
          class="text-tabular text-right"
        >
          <div>
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            >
              0
            </span>
            54.80
            <span>
               × 4 weeks
            </span>
            <div
              class="margin-left-05 width-2 display-inline-block"
            >
              (pt)
            </div>
          </div>
        </td>
        <td
          class="text-tabular text-right"
        >
          <span>
            $ 
          </span>
          <span
            style="visibility: hidden;"
          >
            000,
          </span>
          0.00
        </td>
        <td
          class="oec-table__cell--red text-tabular text-right"
        >
          <div
            class="position-relative one-half-char-right"
          >
            (
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            >
              0
            </span>
            2,192.00)
          </div>
        </td>
      </tr>
      <tr>
        <th>
          <strong>
            Total
          </strong>
        </th>
        <td
          class="oec-table__cell--strong oec-table__cell--red"
        >
          8/70 spaces
        </td>
        <td
          class="text-tabular text-right"
        />
        <td
          class="oec-table__cell--strong text-tabular text-right"
        >
          <span>
            $ 
          </span>
          <span
            style="visibility: hidden;"
          />
          4,455.60
        </td>
        <td
          class="oec-table__cell--strong oec-table__cell--red text-tabular text-right"
        >
          <div
            class="position-relative one-half-char-right"
          >
            (
            <span>
              $ 
            </span>
            <span
              style="visibility: hidden;"
            />
            27,167.20)
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <form
    autocomplete="off"
    class="usa-form"
    novalidate=""
  >
    <h2>
      Other Revenue
    </h2>
    <fieldset
      aria-required="true"
      class="grid-gap grid-row usa-fieldset"
      id="other-revenue"
    >
      <legend
        id="fieldset-legend-other-revenue"
      >
        <span
          class="usa-sr-only"
        >
          Other Revenue
        </span>
      </legend>
      <div
        class="grid-gap grid-row"
      >
        <div
          class="mobile-lg:grid-col-12"
        >
          <div
            class="flex-fill usa-form-group"
          >
            <label
              class="usa-label"
              for="c4k-revenue"
            >
              <span
                class="text-bold"
              >
                Care 4 Kids
              </span>
              <span>
                 (0 kids receiving subsidies)
              </span>
               
            </label>
            <input
              aria-required="true"
              class="usa-input"
              id="c4k-revenue"
              name="c4KRevenue"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="usa-form-group "
        >
          <div
            class="margin-top-3"
          />
          <div
            class="usa-checkbox undefined"
          >
            <input
              class="usa-checkbox__input"
              id="c4k-includes-retroactive-retroactiveC4KRevenue"
              name="retroactiveC4KRevenue"
              type="checkbox"
              value="retroactiveC4KRevenue"
            />
            <label
              class="usa-checkbox__label"
              for="c4k-includes-retroactive-retroactiveC4KRevenue"
            >
              Includes retroactive payment for past months
            </label>
          </div>
        </div>
        <div
          class="mobile-lg:grid-col-12"
        >
          <div
            class="flex-fill usa-form-group"
          >
            <label
              class="usa-label"
              for="family-fees-revenue"
            >
              <span
                class="text-bold"
              >
                Family Fees
              </span>
               
            </label>
            <input
              aria-required="true"
              class="usa-input"
              id="family-fees-revenue"
              name="familyFeesRevenue"
              type="text"
              value="$1,000.00"
            />
          </div>
        </div>
        <div
          class="mobile-lg:grid-col-12"
        >
          <div
            class=" usa-form-group"
          >
            <label
              class="usa-label"
              for="cdc-report-comment"
            >
              <span
                class="text-bold"
              >
                Anything to share with the Office of Early Childhood about your report?
              </span>
               
              <span
                class="usa-hint"
              >
                 (optional)
              </span>
            </label>
            <textarea
              aria-required="false"
              class="usa-textarea"
              id="cdc-report-comment"
              name="comment"
            />
          </div>
        </div>
      </div>
    </fieldset>
    <button
      class="usa-button"
      type="button"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
